src/Shared/RequestDataHelper.cs

```diff
@@ -84,7 +87,27 @@ public void SetActivityDisplayName(Activity activity, string originalHttpMethod,
         var normalizedHttpMethod = this.GetNormalizedHttpMethod(originalHttpMethod);
         var namePrefix = normalizedHttpMethod == "_OTHER" ? "HTTP" : normalizedHttpMethod;
 
-        activity.DisplayName = string.IsNullOrEmpty(httpRoute) ? namePrefix : $"{namePrefix} {httpRoute}";
+        activity.DisplayName = GetDisplayName();
+
+        string GetDisplayName()
+        {
+            if (string.IsNullOrEmpty(httpRoute))
+            {
+                return namePrefix;
+            }
+
+            if (DisplayNameCache.TryGetValue(httpRoute!, out var displayName))
+            {
+                return displayName;
+            }
+
+            if (DisplayNameCache.Count < DisplayNameCacheSize)
+            {
+                return DisplayNameCache.GetOrAdd(httpRoute!, $"{namePrefix} {httpRoute}");
```

Comment:
Re-checking - Can two http methods have the same route values. if that happens, we will incorrectly set the display name here.

e.g. 
Put /api/document/{id}
Get api/document/{id}